File: infisocial_flutter/lib/models/user_model.dart

enum Gender { male, female, other }

class UserModel {
  final String? id;
  final String firstName;
  final String lastName;
  final String username;
  final String email;
  final String? bio;
  final String? avatarUrl;
  final String? age;
  final String? gender;
  final List<String> followers;
  final List<String> following;

  UserModel({
    this.id,
    required this.firstName,
    required this.lastName,
    required this.username,
    required this.email,
    this.bio,
    this.avatarUrl,
    this.age,
    this.gender,
    this.followers = const [],
    this.following = const [],
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    final _user = UserModel(
      id: json['_id'].toString(),
      firstName: json['firstName'],
      lastName: json['lastName'] ?? '',
      username: json['username'],
      email: json['email'],
      bio: json['bio'] ?? '',
      avatarUrl: json['avatarUrl'] ?? '',
      age: json['age'] != null ? json['age'].toString() : '',
      gender: json['gender'] ?? '',
      followers: List<String>.from(json['followers']),
      following: List<String>.from(json['following']),
    );
    return _user;
  }

  Map<String, dynamic> toJson() {
    return {
      "_id": id,
      "firstName": firstName,
      "lastName": lastName,
      "username": username,
      "email": email,
      "bio": bio,
      "avatarUrl": avatarUrl,
      "age": age,
      "gender": gender,
      "followers": followers,
      "following": following,
    };
  }
}

File: infisocial_node/layers/utils/models/user.model.js
const mongoose = require("mongoose");

const userSchema = new mongoose.Schema({
    username: {
        type: String,
        required: true,
        unique: true,
        index: true,
        lowercase: true,
        trim: true
    },
    email: {
        type: String,
        required: true,
        unique: true,
        lowercase: true,
        trim: true
    },
    firstName: {
        type: String,
        required: true,
        index: true
    },
    lastName: {
        type: String,
        index: true
    },
    age: {
        type: Number,
        min: 0,
        max: 150,
        set: (v) => Math.floor(v),
    },
    gender: {
        type: String,
        enum: ['male', 'female', 'other']
    },
    bio: {
        type: String,
        default: ""
    },
    avatarUrl: {
        type: String,
        default: ""
    },
    deviceToken: {
        type: String,
        default: null
    },
    followers: {
        type: [mongoose.Schema.Types.ObjectId],
        ref: "User",
        default: []
    },
    following: {
        type: [mongoose.Schema.Types.ObjectId],
        ref: "User",
        default: []
    },
    isActive: {
        type: Boolean,
        default: true
    },
}, { timestamps: true });

const User = mongoose.model("User", userSchema);

module.exports = User;
